apps/api/src/app/controllers/command/command.controller.ts

```diff
@@ -20,8 +33,20 @@ import validateCommandFSAType from './command-fsa/validateCommandFSAType';
 export class CommandController {
     constructor(private readonly commandHandlerService: CommandHandlerService) {}
 
+    @ApiBearerAuth('JWT')
+    @UseGuards(AdminJwtGuard)
     @Post('')
-    async executeCommand(@Res() res, @Body() commandFSA: CommandFSA) {
+    async executeCommand(@Request() req, @Res() res, @Body() commandFSA: CommandFSA) {
+        const { user } = req;
+
+        if (!user || !(user instanceof CoscradUserWithGroups)) {
+            throw new UnauthorizedException();
+        }
+
+        if (!user.isAdmin()) {
+            throw new UnauthorizedException();
```

Comment:
Can we do this in a guard instead?

Reply:
[here](https://www.pivotaltracker.com/story/show/182785593) is a story for that.